pydis_core/utils/error_handling/commands/manager.py

```diff
@@ -0,0 +1,61 @@
+from collections.abc import Callable
+from typing import NoReturn
+
+from discord import Interaction
+from discord.ext.commands import Context
+
+from pydis_core.utils.error_handling.commands import AbstractCommandErrorHandler
+from pydis_core.utils.logging import get_logger
+
+log = get_logger(__name__)
+
+
+class CommandErrorManager:
+    """A class that registers error handlers and handles all command related errors."""
+
+    def __init__(self, default: AbstractCommandErrorHandler, handlers: list[AbstractCommandErrorHandler] | None = None):
+        self.handlers = handlers or []
+        self.registered_handlers = set(type(handler).__name__ for handler in self.handlers)
+        self.default = default
+
+    async def handle_error(
+        self,
+        error: Exception,
+        context_or_interaction: Context | Interaction
+    ) -> None:
+        """
+        Handle a discord exception.
+
+        This works simply by iterating over the available handlers, and chooses the one capable of handling the error.
```

Comment:
This reads awkardly but I dunno the proper grammar terms to describe it. Should be "chooses" -> "choosing", but I think this and the following line can be rewritten more tersely as

> Iterate through available handlers by registration order, and choose the first one capable of handling the error as determined by `should_handle_error`; there is no priority system.

Reply:
I just went ahead and applied your suggestio, sounded good enough to me
bd62ce8